### PR TITLE
#1252 added RemoteNodeRestartGateSpec

### PR DESCRIPTION
**Pull request by maxim-s**

### CODE REVIEW

**Comment by Aaronontheweb** [comment]
Ah, it says this commit is 29 days old - that can happen sometimes during a rebase.

http://stackoverflow.com/a/31540373/377476 - use this to update the commit timestamp to now and TeamCity will pick it up and run the usual CI steps.

**Comment by Aaronontheweb** [comment]
@maxim-s looks like this spec failed on Node 1:

```
[Node1][FAIL] Akka.Remote.Tests.MultiNode.RemoteNodeRestartGateMultiNode1.MustAllowRestartedNodeToPassThroughGate
[Node1][FAIL-EXCEPTION] Type: Xunit.Sdk.TrueException
--> [Node1][FAIL-EXCEPTION] Message: Failed: Timeout 00:00:09.9936719 while waiting for a message of type Akka.Actor.ActorIdentity 
Expected: True
Actual:   False
--> [Node1][FAIL-EXCEPTION] StackTrace:    at Xunit.Assert.True(Nullable`1 condition, String userMessage) in c:\TeamCity\buildAgent\work\74856245f07a90f0\src\xunit.assert\Asserts\BooleanAsserts.cs:line 90
   at Akka.TestKit.Xunit2.XunitAssertions.Fail(String format, Object[] args) in D:\BuildAgent\work\d26c9d7f36545acd\src\contrib\testkits\Akka.TestKit.Xunit2\XunitAssertions.cs:line 21
   at Akka.TestKit.TestKitBase.InternalExpectMsgEnvelope[T](Nullable`1 timeout, Action`2 assert, String hint, Boolean shouldLog) in D:\BuildAgent\work\d26c9d7f36545acd\src\core\Akka.TestKit\TestKitBase_Expect.cs:line 193
   at Akka.TestKit.TestKitBase.InternalExpectMsgEnvelope[T](Nullable`1 timeout, Action`1 msgAssert, Action`1 senderAssert, String hint) in D:\BuildAgent\work\d26c9d7f36545acd\src\core\Akka.TestKit\TestKitBase_Expect.cs:line 179
   at Akka.TestKit.TestKitBase.InternalExpectMsg[T](Nullable`1 timeout, Action`1 msgAssert, String hint) in D:\BuildAgent\work\d26c9d7f36545acd\src\core\Akka.TestKit\TestKitBase_Expect.cs:line 154
   at Akka.TestKit.TestKitBase.ExpectMsg[T](Nullable`1 duration, String hint) in D:\BuildAgent\work\d26c9d7f36545acd\src\core\Akka.TestKit\TestKitBase_Expect.cs:line 29
   at Akka.Remote.Tests.MultiNode.RemoteNodeRestartGateSpec.<>c__DisplayClass6_0.<MustAllowRestartedNodeToPassThroughGate>b__1() in D:\BuildAgent\work\d26c9d7f36545acd\src\core\Akka.Remote.Tests.MultiNode\RemoteNodeRestartGateSpec.cs:line 70
   at Akka.TestKit.TestKitBase.Within[T](TimeSpan min, TimeSpan max, Func`1 function, String hint) in D:\BuildAgent\work\d26c9d7f36545acd\src\core\Akka.TestKit\TestKitBase_Within.cs:line 79
   at Akka.TestKit.TestKitBase.Within[T](TimeSpan max, Func`1 function) in D:\BuildAgent\work\d26c9d7f36545acd\src\core\Akka.TestKit\TestKitBase_Within.cs:line 51
   at Akka.Remote.Tests.MultiNode.RemoteNodeRestartGateSpec.<MustAllowRestartedNodeToPassThroughGate>b__6_0() in D:\BuildAgent\work\d26c9d7f36545acd\src\core\Akka.Remote.Tests.MultiNode\RemoteNodeRestartGateSpec.cs:line 66
   at Akka.Remote.TestKit.MultiNodeSpec.RunOn(Action thunk, RoleName[] nodes) in D:\BuildAgent\work\d26c9d7f36545acd\src\core\Akka.Remote.TestKit\MultiNodeSpec.cs:line 500
   at Akka.Remote.Tests.MultiNode.RemoteNodeRestartGateSpec.MustAllowRestartedNodeToPassThroughGate() in D:\BuildAgent\work\d26c9d7f36545acd\src\core\Akka.Remote.Tests.MultiNode\RemoteNodeRestartGateSpec.cs:line 54
```

**Comment by maxim-s** [comment]
I have found problem, will be silved asap

**Comment by maxim-s** [comment]
@Aaronontheweb thank you

**Comment by Aaronontheweb** [comment]
@maxim-s did you ever fix the issue here?

File: src/core/Akka.Remote.Tests.MultiNode/RemoteNodeRestartGateSpec.cs [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+
+        private IActorRef Identify(RoleName role, string actorName)
+        {
+            Sys.ActorSelection(Node(role)/"user"/actorName).Tell(new Identify(actorName));
+            return ExpectMsg<ActorIdentity>().Subject;
+        }
+
+
+        [MultiNodeFact]
+        public void MustAllowRestartedNodeToPassThroughGate()
```
Comment (Silv3rcircl3) [inline src/core/Akka.Remote.Tests.MultiNode/RemoteNodeRestartGateSpec.cs]: Regarding to #1510 could you maybe change the method name to underscore format?

**Comment by maxim-s** [comment]
@Aaronontheweb in progress, continue digging

**Comment by maxim-s** [comment]
just synchronization, still should be failed

**Comment by maxim-s** [comment]
@Aaronontheweb failed again on Node2

File: src/core/Akka.Remote.Tests.MultiNode/RemoteNodeRestartGateSpec.cs [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+    public class RemoteNodeRestartGateSpecConfig : MultiNodeConfig
+    {
+        public RemoteNodeRestartGateSpecConfig()
+        {
+            First = Role("first");
+            Second = Role("second");
+
+            CommonConfig = new Config(DebugConfig(false), ConfigurationFactory.ParseString(
+                @"akka.loglevel = DEBUG
+                   akka.remote.log-remote-lifecycle-events = INFO
```
Comment (Danthar) [inline src/core/Akka.Remote.Tests.MultiNode/RemoteNodeRestartGateSpec.cs]: wrong value for  akka.remote.log-remote-lifecycle-events should be 'on' or 'off'

Comment (maxim-s) [inline src/core/Akka.Remote.Tests.MultiNode/RemoteNodeRestartGateSpec.cs]: why if default is "DEBUG" ?  https://github.com/akkadotnet/akka.net/blob/dev/src/core/Akka.Remote/RemoteSettings.cs#L38

**Comment by alexvaluyskiy** [comment]
@maxim-s could you rebase it again?

File: src/core/Akka.Remote.Tests.MultiNode/RemoteNodeRestartGateSpec.cs [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+
+                    {
+                        freshSystem.ActorSelection(new RootActorPath(firstAddress)/"user"/"subject")
+                            .Tell(new Identify("subject"), probe);
+                        probe.ExpectMsg<ActorIdentity>();
+                    });
+                });
+
+                // Now the other system will be able to pass, too
+                freshSystem.ActorOf(Props.Create(() => new Subject()), "subject");
```
Comment (Aaronontheweb) [inline src/core/Akka.Remote.Tests.MultiNode/RemoteNodeRestartGateSpec.cs]: Setting a breakpoint here allows the spec to pass... Interesting. I'll keep looking into it.

Comment (Aaronontheweb) [inline src/core/Akka.Remote.Tests.MultiNode/RemoteNodeRestartGateSpec.cs]: Hah. figured it out - minor mistake on the line below.

**Comment by Aaronontheweb** [comment]
@maxim-s found two bugs in the spec and fixed them - replaced in https://github.com/akkadotnet/akka.net/pull/2120

**Pull request closed without merging**